contrib/initramfs/scripts/zfs

```diff
@@ -1,3 +1,5 @@
+#!/bin/sh
+
```

Comment:
Why are you adding a shebang? I don't think this is something meant to be directly executed? (After all, it seems to be working now without one.)

Reply:
It is not strictly needed as this file is meant to be sourced rather than executed. Adding explicit interpreter would help readers and static analysis tools (like shellcheck) determine the desired syntax. It has no effect on sourced files - it would be treated as comment.

Reply:
Generally speaking, I would prefer not to add shebangs to files which are sourced. I have some experience where this was more of an issue on another project differentiating Python scripts which are executed from Python library files which are imported. But I'll mark this approved and leave the decision to you (well, and ultimately Brian when he decides about merging this).

Reply:
Thanks, Richard, for understanding. Would you elaborate on Python problem with shebangs? An example would help.

Reply:
It was a different project, not OpenZFS. There was a bunch of discussion around shebangs and different Python versions and where different distro a package them. That issue isn’t directly applicable here or even to shell in general (since the shell shebang is standard). But I do think there is some value in differentiating what is executed vs sourced. If someone comes along later and sees a shebang in a file that is installed without execute permission, for example, that might lead to some confusion. That said, I do see your point about things like shellcheck. So I think reasonable people could go either way. A possible middle ground is to keep a shebang but put a comment below it that the script is not executed and the shebang is for shellcheck.

Reply:
I see you point and I do agree that some sort of differentiation is needed. I removed the shebang since it's not relevant to this correction anyways.